OctoKit/NotificationThread.swift

```diff
@@ -0,0 +1,336 @@
+import Foundation
+import RequestKit
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+// TODO add enum for type and reason
+
+// MARK: - Model
+
+open class NotificationThreadSubject: Codable {
+    open internal(set) var id: Int = -1
+    open var title: String?
+    open var url: String?
+    open var latestCommentUrl: String?
+    open var type: String?
+    
+    enum CodingKeys: String, CodingKey {
+        case title
+        case url
+        case latestCommentUrl = "latest_comment_url"
+        case type
+    }
+}
+
+open class NotificationThread: Codable {
+    open internal(set) var id: String? = "-1"
+    open var unread: Bool?
+    open var reason: String?
+    open var updatedAt: Date?
+    open var lastReadAt: Date?
+    open private(set) var subject = NotificationThreadSubject()
+    open private(set) var repository = Repository()
+    
+    enum CodingKeys: String, CodingKey {
+        case id
+        case unread
+        case reason
+        case updatedAt = "updated_at"
+        case lastReadAt = "last_read_at"
+        case subject
+        case repository
+    }
+}
+
+
+open class ThreadSubscription: Codable {
+    open internal(set) var id: Int? = -1
+    open var subscribed: Bool?
+    open var ignored: Bool?
+    open var reason: String?
+    open var url: String?
+    open var threadUrl: String?
+    
+    enum CodingKeys: String,CodingKey {
+        case id
+        case subscribed
+        case ignored
+        case reason
+        case url
+        case threadUrl = "thread_url"
+    }
+}
+
+
+
+// MARK: - Request
+
+public extension Octokit {
+    
+    
+    /**
+     List all notifications for the current user, sorted by most recently updated.
+     - parameter session: RequestKitURLSession, defaults to URLSession.sharedSession()
+     - parameter all: show notifications marked as read `false` by default.
```

Comment:
`all` for me doesn't sound like what the parameter actually does: filtering by `read` status.